addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -199,19 +200,23 @@ options.registry.gallery = options.Class.extend({
 
         // Dispatch images in columns by always putting the next one in the
         // smallest-height column
-        while (imgs.length) {
-            var min = Infinity;
-            var $lowest;
-            _.each(cols, function (col) {
-                var $col = $(col);
-                var height = $col.is(':empty') ? 0 : $col.find('img').last().offset().top + $col.find('img').last().height() - self.$target.offset().top;
-                if (height < min) {
-                    min = height;
-                    $lowest = $col;
-                }
-            });
-            $lowest.append(imgs.shift());
-        }
+        this.lastModeUpdate = new Promise(async resolve => {
+            for (const img of imgs) {
+                let min = Infinity;
+                let $lowest;
+                _.each(cols, function (col) {
+                    var $col = $(col);
+                    var height = $col.is(':empty') ? 0 : $col.find('img').last().offset().top + $col.find('img').last().height() - self.$target.offset().top;
+                    if (height < min) {
+                        min = height;
+                        $lowest = $col;
+                    }
+                });
+                $lowest.append(img);
+                await wUtils.onceAllImagesLoaded(this.$target);
```

Comment:
I wonder if this await for all images loaded could be done only once outside of the loop :thinking:

Reply:
It has to be inside the loop because at each iteration we want to know the correct height of every column to properly determine which is the smallest one.